tests/blockchain/eth/clients/test_keystore_signer.py

```diff
@@ -84,34 +74,32 @@ def unknown_address():
 
 
 def test_blank_keystore_uri():
-    with pytest.raises(Signer.InvalidSignerURI) as error:
+    with pytest.raises(Signer.InvalidSignerURI, match=
+        'Blank signer URI - No keystore path provided') as error:
         Signer.from_signer_uri(uri='keystore://')  # it's blank!
-    assert 'Blank signer URI - No keystore path provided' in str(error)
 
 
-def test_invalid_keystore(mocker):
-    mock_stat_scandir(mocker)
+def test_invalid_keystore(mocker, tmp_path):
+    with pytest.raises(Signer.InvalidSignerURI) as e:
+        Signer.from_signer_uri(uri=f'keystore:{tmp_path/"nonexistent"}')
 
-    # mock Keystoresigner.__read_keyfile
-    # Invalid keystore values and exception handling
-    mock_keyfile_reader = mocker.patch.object(KeystoreSigner, '_KeystoreSigner__read_keyfile', autospec=True)
-    mock_keyfile_reader.return_value = '0xdeadbeef', dict()
+    empty_path = tmp_path / 'empty_file'
+    open(empty_path, 'x+t').close()
+    with pytest.raises(KeystoreSigner.InvalidKeyfile, match=
+        'Invalid JSON in keyfile at') as e:
+        Signer.from_signer_uri(uri=f'keystore:{empty_path}')
 
-    #
-    # 1 - Create
-    #
-
-    with pytest.raises(KeystoreSigner.InvalidKeyfile) as e:
-        Signer.from_signer_uri(uri=MOCK_KEYSTORE_URI)
-    assert "does not contain a valid ethereum address" in str(e.value)
-
-    # Invalid keyfiles
-    for exception in (FileNotFoundError, KeyError):
-        with pytest.raises(KeystoreSigner.InvalidKeyfile):
-            mock_keyfile_reader.side_effect = exception
-            Signer.from_signer_uri(uri=MOCK_KEYSTORE_URI)
-    mock_keyfile_reader.side_effect = None  # clean up this mess
+    empty_json = tmp_path / 'empty_json'
+    json.dump({}, open(empty_json, 'x+t'))
+    with pytest.raises(KeystoreSigner.InvalidKeyfile, match=
+        'Keyfile does not contain address field at') as e:
+        Signer.from_signer_uri(uri=f'keystore:{empty_json}')
 
+    bad_address = tmp_path / 'bad_address'
+    json.dump({'address':''}, open(bad_address, 'x+t'))
+    with pytest.raises(KeystoreSigner.InvalidKeyfile, match=
+        'does not contain a valid ethereum address') as e:
```

Comment:
:sunglasses: - I have been missing this kwarg from my life!

Reply:
Yes - `match` looks like just what the doctor ordered for this kind of thing. :)